Allow recording while other media playing

cark1 opened the pull request:
Using AVAudioSessionCategoryPlayAndRecord is possibile to have other media playing (for example a muted video).

If we have another active media (for example a video) and we try to terminate the AVAudioSession we have an error.
And if we have an error, rejectStop is called and we lose the recording.
We could just try to deactivate the AVAudioSession, but if this is not possible, we shouldn't make fail the recording entirely.

cark1 committed:
Allow recording while other media playing

Using AVAudioSessionCategoryPlayAndRecord is possibile to have other media playing (for example a muted video).

If we have another active media (for example a video) and we try to terminate the AVAudioSession we have an error.
And if we have an error, rejectStop is called and we lose the recording.
We could just try to deactivate the AVAudioSession, but if this is not possible, we shouldn't make fail the recording entirely.

kevinresol commented:
Completely ignoring the errors doesn't look good to me. Is there any ways to distinguish the different scenarios and still emit the error when appropriate?

cark1 commented:
Yeah, you are totally right.
We could skip the error only when we have this specific error code: [AVAudioSessionErrorCodeIsBusy](https://developer.apple.com/documentation/coreaudiotypes/avaudiosessionerrorcode/avaudiosessionerrorcodeisbusy)

cark1 committed:
skip AVAudioSessionErrorCodeIsBusy error

Now we skip the "session setActive:NO" error only if the error is AVAudioSessionErrorCodeIsBusy (some other media is using the session).

kevinresol commented:
Thanks!

cark1 commented:
Hi @kevinresol , any chance to have a new release on npm? Thank you!

kevinresol commented:
Released 1.4.0

cark1 commented:
> Released 1.4.0

Thank you!
